docs/guides/transactions/running-background-jobs.md

```diff
@@ -178,48 +178,59 @@ Handle.GET("/Hello", () =>
 
 ```csharp
-// Create a database object on the Starcounter thread
-var person = Db.Transact(() => new Person());
+[Database]
+public class Person {
+    public String Id { get; set; }
+}
+class Program {
 
-// The await keyword lets .NET pick the thread
-await Scheduling.RunTask(() => Thread.Sleep(2000));
+    static void Main() {
 
-// .NET might pick a .NET thread or Starcounter thread. If a 
-// .NET thread is picked, this code will throw an exception
-Db.Transact(() => person.Name = "John");
-```
+        Handle.GET("/hello/{?}", (Request req, Int32 id) => {
 
-To prevent undeterministic exceptions like this, use the `Wait` method to wait for the task to finish execution.
+            // As we are on Starcounter thread already, we can create a database object.
+            // Notice that we use `Db.TransactAsync` here. Read about it in a separate article.
+            Db.TransactAsync(() => new Person() { Id = id.ToString() });
 
-```csharp
-var person = Db.Transact(() => new Person());
+            // Now we run an async task that will return the response later.
+            DoSomethingAsync(req, id);
+
+            // Indicating that HTTP response will be returned later.
+            return HandlerStatus.Handled;
+        });
+    }
 
-// Wait for the task to finish with the Wait method
-Scheduling.RunTask(() => Thread.Sleep(2000)).Wait();
+    // Async call that does some await and performs database operations after.
+    static async void DoSomethingAsync(Request req, Int32 id) {
 
-// This is now guaranteed to be on a Starcounter thread
-Db.Transact(() => person.Name = "John");
-```
+        // Now using async Simulating some long operation using await statement.
+        await Task.Delay(10000);
 
-You can also use the task returned by `RunTask` to wait for the task when the Starcounter thread is needed.
+        // Since this is not a Starcounter thread now, so we need to schedule
+        // a Starcounter task so we can do our database operations.
+        await Scheduling.RunTask(() => {
 
-```csharp
-var person = Db.Transact(() => new Person());
-Task task = Scheduling.RunTask(() => Thread.Sleep(2000));
+            // Quering the person.
+            var person = Db.SQL<Person>("SELECT p FROM Person p WHERE p.Id = ?", id.ToString()).FirstOrDefault();
 
-// Call a method that doesn't require a Starcounter thread
-SomeLongCalculation();
+            // Doing some database operations.
+            // Notice that we use `Db.TransactAsync` here. Read about it in a separate article.
+            Db.TransactAsync(() => person.Id += id);
```

Comment:
What's the purpose of using `TransactAsync` here?

Reply:
From the text: "Note also the use of `Db.TransactAsync` - it's needed so that the Starcounter scheduler does not create additional threads." There is a separate article on `Db.TransactAsync`